File: ItalianRestaurantFrontend/src/app/models/meal.ts
import {CategoryDto} from "./categoryDto";

export class Meal {
  id: number
  name: string
  name_pl: string
  image: string
  description: string
  description_pl: string
  price: number
  mealCategory: CategoryDto

  constructor(id: number, name: string, name_pl:string, image: string, description: string, description_pl:string, price: number, mealCategory: CategoryDto) {
    this.id = id
    this.name = name;
    this.name_pl = name_pl;
    this.image = image;
    this.description = description;
    this.description_pl = description_pl;
    this.price = price;
    this.mealCategory = mealCategory;
  }

}

File: ItalianRestaurantBackend/src/main/java/com/example/italianrestaurant/meal/Meal.java
package com.example.italianrestaurant.meal;

import com.example.italianrestaurant.meal.mealcategory.MealCategory;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.util.Objects;

@Entity
@Table(name = "meals")
@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Meal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String name_pl;
    private String image;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(nullable = false)
    @ToString.Exclude
    private MealCategory mealCategory;
    private String description;
    private String description_pl;
    @Column(nullable = false)
    private double price;


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Meal meal = (Meal) o;
        return getId() != null && Objects.equals(getId(), meal.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
